### PR TITLE
Aoc/642 track init end

### PR DESCRIPTION
**Pull request by chengoconnell**
image

## Types of changes

- [ ] Bug fix (a non-breaking change which fixes an issue)
- [x] New feature (a non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly. (PR: #)
- [ ] All new and existing tests are passing.

### CODE REVIEW

File: packages/init/src/index.ts [hunk 1]
```diff
@@ -104,7 +105,12 @@ init()
   .then(() => {
     process.exit(0);
   })
-  .catch((error) => {
+  .catch(async (error) => {
+    await Tracker.get().trackInitEnd(
```
Comment (MarcMcIntosh) [inline packages/init/src/index.ts]: maybe put this after logging?

File: packages/init/src/utils/tracker.ts [hunk 2]
```diff
@@ -6,7 +6,7 @@ export enum EventType {
   DownloadLibrary = "SliceMachine Download Library",
   InitStart = "SliceMachine Init Start",
   InitIdentify = "SliceMachine Init Identify",
-  InitDone = "SliceMachine Init Done",
+  InitEnd = "SliceMachine Init End",
```
Comment (MarcMcIntosh) [inline packages/init/src/utils/tracker.ts:9]: event name will need to be updated in segment :)

**Review by MarcMcIntosh — COMMENTED** [review]
I'll get back to you on how to test this, as the tracking library is a bit weird and we may have to mock it

File: packages/init/src/utils/tracker.ts [hunk 3]
```diff
@@ -114,8 +114,12 @@ export class InitTracker {
     return this._trackEvent(EventType.InitStart, { repo: repoDomain });
   }
 
-  trackInitDone(framework: Models.Frameworks): Promise<void> {
-    return this._trackEvent(EventType.InitDone, { framework });
+  trackInitEnd(
+    framework: Models.Frameworks,
+    success: boolean,
+    error?: string
```
Comment (MarcMcIntosh) [inline packages/init/src/utils/tracker.ts]: add `repo: string` here :)

**Review by MarcMcIntosh — APPROVED** [review]

File: packages/init/__tests__/configure-project.test.ts [hunk 4]
```diff
@@ -260,7 +260,7 @@ describe("configure-project", () => {
 
     expect(successFn).not.toHaveBeenCalled();
     expect(failFn).toHaveBeenCalled();
-    expect(MockTracker).not.toHaveBeenCalled();
+    expect(MockTracker).toHaveBeenCalled();
```
Comment (sintinilorenzo) [inline packages/init/__tests__/configure-project.test.ts]: can you also assert the payload it sent? 
See how it's done in the file below:
```expect(MockTracker).toHaveBeenCalledTimes(1);
    expect(MockTracker.mock.calls[0][0]).toEqual({
      anonymousId: "uuid",
      event: "SliceMachine Init End",
      properties: {
        framework: Models.Frameworks.next,
        repo: "repoName",
        result: "success",
      },
      context: { groupId: { Repository: "repoName" } },
    });```

Comment (sintinilorenzo) [reply packages/init/__tests__/configure-project.test.ts]: or at least test the event and properties: 
```expect(MockTracker.mock.calls[0][0].event).toEqual(...))```
```expect(MockTracker.mock.calls[0][0]. properties).toEqual(...))```

File: packages/init/__tests__/configure-project.test.ts [hunk 5]
```diff
@@ -233,7 +233,7 @@ describe("configure-project", () => {
...
```
Comment (sintinilorenzo) [inline packages/init/__tests__/configure-project.test.ts]: same as other comment

File: packages/init/__tests__/configure-project.test.ts [hunk 6]
```diff
@@ -207,7 +207,7 @@ describe("configure-project", () => {
...
```
Comment (sintinilorenzo) [inline packages/init/__tests__/configure-project.test.ts]: same as other comment

File: packages/init/__tests__/configure-project.test.ts (hunk 4 above)
Comment (chengoconnell) [reply packages/init/__tests__/configure-project.test.ts]: This is already tested in `tracker.test.ts`. Do you think it's still necessary?

Comment (chengoconnell) [reply packages/init/__tests__/configure-project.test.ts]: updated

**Pull request merged**